maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java

```diff
@@ -317,19 +321,78 @@ public void promptForJavaHomeIfMissing(Server server) {
             String path = promptForMissingValueWithOptions(SDKConstants.OPENMRS_SDK_JDK_OPTION,
                     server.getJavaHome(), "path", paths, SDKConstants.OPENMRS_SDK_JDK_CUSTOM, null);
 
+            String requiredJdkVersion;
+            if (server.getPlatformVersion().charAt(0) == 1) {
+                requiredJdkVersion = "1.7";
+            }
+            else {
+                requiredJdkVersion = "1.8";
+            }
+
+            // Use default JAVA_HOME
             if (path.equals(paths.get(0))) {
-                // Use default JAVA_HOME
-                server.setJavaHome(null);
+                if (System.getProperty("java.version").startsWith(requiredJdkVersion)) {
+                    server.setJavaHome(null);
+                }
+                else if (System.getProperty("java.version").startsWith(requiredJdkVersion)) {
+                    server.setJavaHome(null);
+                }
+                else {
+                    showMessage("Your JAVA_HOME version doesn't fit platform requirements:");
+                    showMessage("JAVA_HOME version: " + System.getProperty("java.version"));
+                    showMessage("Required: " + requiredJdkVersion);
+                    promptForJavaHomeIfMissing(server);
```

Comment:
Let's make it less strict and require only java 8 for platform 2.x, but allow using java 8 for platform 1.x with additional confirmation from the user "It is not recommended to run OpenMRS platform {versionHere} on JDK 8. Would you like to select the recommended JDK 7 instead? [Y/n]:"